src/plugins/image/ImageEditor.tsx

```diff
@@ -126,6 +128,17 @@ export function ImageEditor({ src, title, nodeKey, width, height }: ImageEditorP
     [editor, setSelected]
   )
 
+  React.useEffect(() => {
+    if (imagePreviewHandler) {
+      const callPreviewHandler = async () => {
```

Comment:
I think that the handler should be awaited before rendering, otherwise the image will do two network requests - one for the actual `src` and one for the updated one. Hope this makes sense.

Reply:
Ah, good spot. What would you recommend to tackle that? I suppose `useEffect` won't work here?

Reply:
Got it, thank you, the double fetching in `ImageEditor` should be gone now.

Line 17 of `LexicalImageVisitor.ts` is also causing the browser to fetch the original URL, I'll see if there's a way around that.

![Screenshot 2023-09-04 at 18 32 03](https://github.com/mdx-editor/editor/assets/98081005/074b814f-4b0d-4650-ae91-af0e32cf455a)

Specifically where the `src` is assigned to the `new Image()`: `img.src = lexicalNode.getSrc()`

Reply:
The point about the LexicalImageVisitor is a valid one. The reason why I construct the Image there is to grab its HTML while sparing me the hassle of constructing HTML with strings. Here's one thing (I am not sure if it will work, and I am not sure if it does not needs some additional URL sanitizing.

```js
        value: img.outerHTML.replace(/>$/, `src="${lexicalNode.getSrc()}" />`)
```

Reply:
Good suggestion, this seems to work well from my testing. I've added a space before the `src="${lexicalNode.getSrc()}"` for cosmetics, but this resolves the double fetching, and appears as expected in the markdown view:

![Screenshot 2023-09-05 at 09 03 43](https://github.com/mdx-editor/editor/assets/98081005/63238b2b-60df-4135-912b-90dd97b162d8)